iceoryx_posh/include/iceoryx_posh/internal/roudi/service_registry.hpp

```diff
@@ -80,6 +81,14 @@ class ServiceRegistry
               const cxx::optional<capro::IdString_t>& instance,
               const cxx::optional<capro::IdString_t>& event) const noexcept;
 
+    /// @copydoc ServiceDiscovery::findService
+    void find(const cxx::optional<capro::IdString_t>& service,
+              const cxx::optional<capro::IdString_t>& instance,
+              const cxx::optional<capro::IdString_t>& event,
+              cxx::function_ref<void(const ServiceDescriptionEntry&)> function) const noexcept;
```

Comment:
Nitpick: `callable` is more general than `function`, think e.g. of functors
```suggestion
              cxx::function_ref<void(const ServiceDescriptionEntry&)> callable) const noexcept;
```

Reply:
In a mathematical sense function is general enough. And functors in C++ are not even functors (but functions with state). Either is fine here, cf. https://en.wikipedia.org/wiki/Map_(higher-order_function) for concepts.

Reply:
I would prefer the name `callable` as this term is used [throughout the C++ standard](https://eel.is/c++draft/func.def#def:type,callable) and we should be as close to the standard as possible. No hard feelings, though.